include/vikunja/access/PolicyBasedBlockStrategy.hpp

```diff
@@ -0,0 +1,309 @@
+        ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE auto operator--() -> PolicyBasedBlockStrategy&
+        {
+            this->m_index -= this->m_step;
+            return *this;
+        }
+
+        //-----------------------------------------------------------------------------
+        //! Returns the current index and decrements the internal pointer to the
+        //! previous one.
+        //!
+        //! Returns a reference to the current index.
+        ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE auto operator--(int) -> PolicyBasedBlockStrategy
+        {
+            auto ret(*this);
+            this->m_index -= this->m_step;
+            return ret;
+        }
+
+        //-----------------------------------------------------------------------------
+        //! Returns the index + a supplied offset.
+        //!
+        //! \param n The offset.
+        ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE auto operator+(uint64_t n) const -> PolicyBasedBlockStrategy
+        {
+            auto ret(*this);
+            ret.m_index += n * m_step;
+            return ret;
+        }
+
+        //-----------------------------------------------------------------------------
+        //! Returns the index - a supplied offset.
+        //!
+        //! \param n The offset.
+        ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE auto operator-(uint64_t n) const -> PolicyBasedBlockStrategy
+        {
+            auto ret(*this);
+            ret.m_index -= n * m_step;
+            return ret;
+        }
+
+        //-----------------------------------------------------------------------------
+        //! Addition assignment.
+        //!
+        //! \param offset The offset.
+        //!
+        //! Returns the current object offset by the offset.
+        ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE auto operator+=(uint64_t offset) -> PolicyBasedBlockStrategy&
+        {
+            this->m_index += offset * this->m_step;
+            return *this;
+        }
+
+        //-----------------------------------------------------------------------------
+        //! Substraction assignment.
+        //!
+        //! \param offset The offset.
+        //!
+        //! Returns the current object offset by the offset.
+        ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE auto operator-=(uint64_t offset) -> PolicyBasedBlockStrategy&
+        {
+            this->m_index -= offset * this->m_step;
+            return *this;
+        }
+    };
+
+    namespace policies
+    {
+        /**
+         * A memory policy for the PolicyBlockBasedIterator that provides grid striding memory access.
+         */
+        struct GridStridingMemAccessPolicy
+        {
+            template<typename TAcc, typename TIdx>
+            ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE static auto getStartIndex(
+                TAcc const& acc,
+                TIdx const& problemSize __attribute__((unused)),
+                TIdx const& blockSize __attribute__((unused))) -> TIdx const
```

Comment:
```suggestion
                TIdx const& /* problemSize */,
                TIdx const& /* blockSize */)) -> TIdx const
```

If you really need this consider using `[[maybe_unused]]` instead to make it portable.